### PR TITLE
feat(confidential-ledger): workign server, read data from guid

### LINKED ISSUE #95: Confidential-Ledger Server setup

### LINKED ISSUE #100: Search function for retrieved cl entries

### PR DESCRIPTION
**Pull request by FergusKidd**
Server setup working locally, use /read/guid to return data from cl for a given guid. fixes #95 fixes #100

### CODE REVIEW

File: confidential-ledger-demo/server/app.py [hunk 1]
```diff
@@ -46,51 +46,40 @@ async def about(request):
...
-    # Until confidential ledger access is granted:
-    with open("sampledata.json") as json_file:
-        data = json.load(json_file)
-
-        if name == data["Meta"]["guid"]:
-            return JSONResponse(data)
-        else:
-            return JSONResponse({"error": "specified car not found"})
-
+    guid = request.path_params["carid"]
```
Comment (Sealjay-clj) [inline confidential-ledger-demo/server/app.py:49]: maybe ID rather than carID? or guid?

File: confidential-ledger-demo/server/app.py [hunk 2]
```diff
@@ -46,51 +46,40 @@ async def about(request):
...
     # get data
     try:
         # TODO test with real connection
         # TODO use carid dependent on data structure
-        latest_data = cl.rpc_get_latest()
+        latest_data = cl.search_entries_guid(guid)
     except:
         errorMessage = "The confidential data connection seems not to be working"
```
Comment (Sealjay-clj) [inline confidential-ledger-demo/server/app.py:56]:
```suggestion
        errorMessage = "The confidential data connection isn't working"
```

File: confidential-ledger-demo/server/confidentialledger.py [hunk 3]
```diff
@@ -1,224 +1,90 @@
-# Copyright (c) 2021 Microsoft.
-# Copyright (c) 2021 Avanade Inc.
+# Import the Azure authentication library
```
Comment (Sealjay-clj) [inline confidential-ledger-demo/server/confidentialledger.py:1]: We do need Avanade & MSFT inc copyrights

File: confidential-ledger-demo/server/confidentialledger.py [hunk 4]
```diff
@@ -1,224 +1,90 @@
...
-from typing import Dict, NamedTuple, Iterable
-import argparse
+from azure.confidentialledger import ConfidentialLedgerClient
+from azure.confidentialledger.identity_service import (
+    ConfidentialLedgerIdentityServiceClient,
+)
 
-from dotenv import load_dotenv
-import requests
 import os
```
Comment (Sealjay-clj) [inline confidential-ledger-demo/server/confidentialledger.py:12]: We should reorder the imports - have you run pylint?

File: confidential-ledger-demo/server/app.py (hunk 1 above)
Comment (FergusKidd) [reply confidential-ledger-demo/server/app.py:49]: changed to guid

File: confidential-ledger-demo/server/app.py (hunk 2 above)
Comment (FergusKidd) [reply confidential-ledger-demo/server/app.py:56]: updated

File: confidential-ledger-demo/server/confidentialledger.py (hunk 3 above)
Comment (FergusKidd) [reply confidential-ledger-demo/server/confidentialledger.py:1]: added

File: confidential-ledger-demo/server/confidentialledger.py (hunk 4 above)
Comment (FergusKidd) [reply confidential-ledger-demo/server/confidentialledger.py:12]: will run and re-factor

**Comment by FergusKidd** [comment]
@Sealjay-clj I have addressed your comments

**Pull request merged**